### ISSUE TITLE
StoreReview: Native rating modal doesn't display on all users devices

### ISSUE THREAD

**Issue by TomBerriot** [expo/expo#20767]
### Summary

Store Review docs: https://docs.expo.dev/versions/latest/sdk/storereview/

Hello, we have a problem with the opening of the app review modal with Expo, that only a minor part of users already well selected seem to see and go through the end of rating process. Many click to see it voluntarily and only a minor part actually publish a review.

We have an application with an app review flow customized to optimize good reviews by targeting specific users already using our app regularly.
Before showing the native app rating modal, we display a previous modal asking if the user want to review us or not. And this previous modal is displayed at the end of a flow of usage of our app, so not disturbing the user up front when he just opened the app.

But actually, we go from 50 logs of clicks to open the native rating modal to 8 review added to our app store / play store in total.

Our app already only supports SDKs recent enough to support native rating modal.   

Would you know what could be failing that much ?

### SDK Version

47.0

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.15 Ubuntu 20.04.4 LTS (Focal Fossa)
      Shell: 5.8 - /usr/bin/zsh
    Binaries:
      Node: 16.15.1 - /usr/bin/node
      Yarn: 1.22.19 - /usr/bin/yarn
      npm: 8.11.0 - /usr/bin/npm
    npmPackages:
      expo: ^47.0.0 => 47.0.3 
      react: 18.1.0 => 18.1.0 
      react-dom: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
    Expo Workflow: managed

### Minimal reproducible example

Here is our code to launch the native rating modal via events coming from our web app in our website (using a webview as a PWA)

```
if (message.nativeEvent.data.startsWith('request-review')) {
    StoreReview.hasAction().then((hasAction) => {
        if (hasAction) {
            StoreReview.requestReview().then((res) => {
                console.log(res)
            }).catch((error) => {
                console.log(error);
            });
        } else {
            const reviewUrl = Platform.OS === 'ios' ? Constants.manifest.extra.ios_review_url : Constants.manifest.extra.android_review_url;
            Linking.openURL(reviewUrl);
        }
    });
}
```

**Comment by maxizhukov** [comment]
I have same issue, for me today AppRating banner was not clickable at all, can not rate or even close and after one hour it stopped to show any rating banner.
```
"expo-store-review": "~6.0.0"
"expo": "^47.0.0"
```

```
const showRating = async () => {
		await StoreReview.requestReview();
	}
```

**Comment by TomBerriot** [comment]
> I have same issue, for me today AppRating banner was not clickable at all

I can't confirm what you say because when we test by ourselves when we click the modal, everything works on our end.

I can only say that a too many large part of users do not seem to go through the end of the process so it must be due to the modal not showing on a large part of devices, despite our conditioning of `StoreReview.hasAction()` to request Store Review

**Comment by maxizhukov** [comment]
@TomBerriot Tried to use it with iPhone 11 Pro Max and iOS 16
Modal was showing before, but was not clickable and now it's not even showing modal to rate. There was no updates in app in that time. 

What I was thinking, that in my case it might be also because of this 2 things:

1. App was only today first time approved by Apple, so maybe storeUrl still can not be find by apple

2. Before we have updated settings for compileSdkVersion and deploymentTarget, so maybe it's not working with 13.0
```
"android": {
            "compileSdkVersion": 33,
            "targetSdkVersion": 33,
            "buildToolsVersion": "33.0.0"
          },
          "ios": {
            "deploymentTarget": "13.0"
          }
```

I have also tried to debug in expoGo app:
StoreReview.hasAction() => returns true
StoreReview.isAvailableAsync() => return true
StoreReview.requestReview() => return undefined
